app/src/main/java/com/seafile/seadroid2/data/DataManager.java

```diff
@@ -929,4 +1023,103 @@ public String search(String query, int page) throws SeafException {
             return null;
         }
     }
+
+    private FileBlocks chunkFile(String encKey, String enkIv, String filePath) throws IOException {
+        File file = new File(filePath);
+        InputStream in = null;
+        OutputStream out = null;
+        byte[] buffer = new byte[BUFFER_SIZE];
+        FileBlocks seafBlock = new FileBlocks();
+        try {
+            in = new FileInputStream(file);
+
+            // Log.d(DEBUG_TAG, "file size " + file.length());
+            while (in.read(buffer, 0, BUFFER_SIZE) != -1) {
```

Comment:
`in.read(buffer, 0, BUFFER_SIZE)` may return less than BUFFER_SIZE bytes.

Better use e.g. https://developer.android.com/reference/java/io/DataInputStream.html#readFully(byte[], int, int)